flocker/ca/test/test_ca.py

```diff
@@ -13,28 +13,299 @@
+    def test_load_error_on_unreadable_certificate_file(self):
+        """
+        A ``PathError`` is raised if the certificate file path given to
+        ``ControlCredential.from_path`` cannot be opened for reading.
+        """
+        path = FilePath(self.mktemp())
+        path.makedirs()
+        crt_path = path.child(CONTROL_CERTIFICATE_FILENAME)
+        crt_file = crt_path.open(b'w')
+        crt_file.write(b"dummy")
+        crt_file.close()
+        # make file unreadable
+        crt_path.chmod(64)
+        key_path = path.child(CONTROL_KEY_FILENAME)
+        key_file = key_path.open(b'w')
+        key_file.write(b"dummy")
+        key_file.close()
+        # make file unreadable
+        key_path.chmod(64)
+        e = self.assertRaises(
+            PathError, ControlCredential.from_path, path
+        )
+        expected = (
+            b"Certificate file could not be opened. "
+            b"Permission denied {path}"
+        ).format(path=crt_path.path)
+        self.assertEqual(str(e), expected)
+
+    @not_root
+    @skip_on_broken_permissions
+    def test_load_error_on_unreadable_key_file(self):
+        """
+        A ``PathError`` is raised if the key file path given to
+        ``ControlCredential.from_path`` cannot be opened for reading.
+        """
+        path = FilePath(self.mktemp())
+        path.makedirs()
+        crt_path = path.child(CONTROL_CERTIFICATE_FILENAME)
+        crt_file = crt_path.open(b'w')
+        crt_file.write(b"dummy")
+        crt_file.close()
+        key_path = path.child(CONTROL_KEY_FILENAME)
+        key_file = key_path.open(b'w')
+        key_file.write(b"dummy")
+        key_file.close()
+        # make file unreadable
+        key_path.chmod(64)
+        e = self.assertRaises(
+            PathError, ControlCredential.from_path, path
+        )
+        expected = (
+            b"Private key file could not be opened. "
+            b"Permission denied {path}"
+        ).format(path=key_path.path)
+        self.assertEqual(str(e), expected)
+
+    def test_certificate_subject_control_service(self):
+        """
+        A cert written by ``ControlCredential.initialize`` has the
+        subject common name "control-service"
+        """
+        cc = ControlCredential.initialize(self.path, self.ca)
+        cert = cc.certificate.original
+        subject = cert.get_subject()
+        self.assertEqual(subject.CN, b"control-service")
+
+    def test_certificate_ou_matches_ca(self):
+        """
+        A cert written by ``ControlCredential.initialize`` has the issuing
+        authority's common name as its organizational unit name.
+        """
+        cc = ControlCredential.initialize(self.path, self.ca)
+        cert = cc.certificate.original
+        issuer = cert.get_issuer()
+        subject = cert.get_subject()
+        self.assertEqual(
+            issuer.CN,
+            subject.OU
+        )
+
+    def test_certificate_is_signed_by_ca(self):
```

Comment:
Not sure this actually proves what it claims to prove. But functional test does provide more coverage.